protocol/app/process/transactions.go

```diff
@@ -4,20 +4,37 @@ import (
 	errorsmod "cosmossdk.io/errors"
 	abci "github.com/cometbft/cometbft/abci/types"
 	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/dydxprotocol/v4-chain/protocol/lib"
+	"slices"
 )
 
-type txtype int
-
 const (
-	ProposedOperationsTxType txtype = 1
-	AcknowledgeBridgesTxType txtype = 2
-	AddPremiumVotesTxType    txtype = 3
-	UpdateMarketPricesTxType txtype = 4
+	minTxsCount                   = 4
+	proposedOperationsTxIndex     = 0
+	updateMarketPricesTxLenOffset = -1
+	addPremiumVotesTxLenOffset    = -2
+	acknowledgeBridgesTxLenOffset = -3
+	lastOtherTxLenOffset          = acknowledgeBridgesTxLenOffset
+	firstOtherTxOffset            = 1
 )
 
-const (
-	MinTxsCount = 4
-)
+func init() {
```

Comment:
Was going to ask "is this function ever called?", but just learned about this special function in golang. Cool!